src/Adapter/Product/Image/Repository/ProductImageMultiShopRepository.php

```diff
@@ -0,0 +1,242 @@
+            ->addOrderBy('i.id_image', 'ASC')
+            ->execute()
+            ->fetchAll()
+        ;
+
+        if (!$results) {
+            return [];
+        }
+
+        $imagesIds = [];
+        foreach ($results as $result) {
+            $imagesIds[] = new ImageId((int) $result['id_image']);
+        }
+
+        return $imagesIds;
+    }
+
+    /**
+     * @param ImageId $imageId
+     *
+     * @return Image
+     *
+     * @throws CoreException
+     */
+    public function get(ImageId $imageId, ShopId $shopId): Image
+    {
+        /** @var Image $image */
+        $image = $this->getObjectModelForShop(
+            $imageId->getValue(),
+            Image::class,
+            ProductImageNotFoundException::class,
+            $shopId
+        );
+
+        return $image;
+    }
+
+    /**
+     * @param ImageId $imageId
+     *
+     * @return ShopId[]
+     */
+    public function getAssociatedShopIds(ImageId $imageId): array
+    {
+        $qb = $this->connection->createQueryBuilder();
+        $qb
+            ->select('id_shop')
+            ->from($this->dbPrefix . 'image_shop')
+            ->where('id_image = :imageId')
+            ->setParameter('imageId', $imageId->getValue())
+        ;
+
+        $result = $qb->execute()->fetchAll();
+        if (empty($result)) {
+            return [];
+        }
+
+        $shops = [];
+        foreach ($result as $shop) {
+            $shops[] = new ShopId((int) $shop['id_shop']);
+        }
+
+        return $shops;
+    }
+
+    public function create(ProductId $productId, ShopConstraint $shopConstraint): Image
+    {
+        $productIdValue = $productId->getValue();
+        $image = new Image();
+        $image->id_product = $productIdValue;
+        $image->cover = !Image::getCover($productIdValue);
+
+        $shopIds = $this->productMultiShopRepository->getShopIdsByConstraint($productId, $shopConstraint);
+        $this->addObjectModelToShops($image, $shopIds, CannotAddProductImageException::class);
+
+        //Todo: Maybe useless try test behat with this
+        try {
+            if (!$image->associateTo($shopIds)) {
+                throw new ProductImageException(sprintf(
+                    'Failed to associate product image #%d with shops',
+                    $image->id
+                ));
+            }
+        } catch (PrestaShopException $e) {
+            throw new CoreException(
+                sprintf('Error occurred when trying to associate image #%d with shops', $image->id),
+                0,
+                $e
+            );
+        }
+
+        return $image;
+    }
+
+    /**
+     * @param Image $image
+     * @param array<int> $shopsToRemoveImageFrom
+     *
+     * @return void
+     */
+    public function deleteFromShops(Image $image, array $shopsToRemoveImageFrom)
+    {
+        $this->deleteObjectModelFromShops($image, $shopsToRemoveImageFrom, CannotDeleteProductImageException::class);
```

Comment:
This is probably where you'll need to assert the presence of the cover and potentially re-assign it